kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -264,16 +264,16 @@ static class InitalizerNetFilter implements NetFilter {
         @Override
         public void selectServer(NetFilter.NetFilterContext context) {
             List<FilterAndInvoker> apiVersionFilters = decodePredicate.isAuthenticationOffloadEnabled() ? List.of()
-                    : FilterAndInvoker.build(apiVersionsIntersectFilter);
+                    : FilterAndInvoker.build("ApiVersionsIntersect (internal)", apiVersionsIntersectFilter);
```

Comment:
The externally configured filters must have names matching a certain regex. Here we need to name the internal filters added by the proxy itself. Obviously we want to make it clear that they're internal, but the exact naming schema is up for debate.